packages/framework-provider-azure/src/library/events-stream-producer-adapter.ts

```diff
@@ -0,0 +1,56 @@
+import { BoosterConfig, EventEnvelope } from '@boostercloud/framework-types'
+import { CreateBatchOptions, EventHubProducerClient } from '@azure/event-hubs'
+import { getLogger } from '@boostercloud/framework-common-helpers'
+import { partitionKeyForEvent } from './partition-keys'
+
+export async function produceEventsStream(
+  producer: EventHubProducerClient,
+  entityName: string,
+  entityID: string,
+  eventEnvelopes: Array<EventEnvelope>,
+  config: BoosterConfig
+): Promise<void> {
+  const logger = getLogger(config, 'events-stream-producer#produceEventsStream')
+  logger.debug('Producing eventEnvelopes', eventEnvelopes)
+  const batchOptions: CreateBatchOptions = {
+    partitionKey: partitionKeyForEvent(entityName, entityID),
+  }
+
+  let batch = await producer.createBatch(batchOptions)
+  let numEventsSent = 0
+  let i = 0
+  while (i < eventEnvelopes.length) {
+    // messages can fail to be added to the batch if they exceed the maximum size configured for the EventHub.
+    const eventEnvelope = eventEnvelopes[i]
+    const isAdded = batch.tryAdd({ body: eventEnvelope })
+
+    if (isAdded) {
+      logger.info(`Added ${JSON.stringify(eventEnvelope)} to the batch`)
+      ++i
+      continue
+    }
+
+    if (batch.count === 0) {
+      throw new Error(`Message was too large and can't be sent until it's made smaller. ${eventEnvelope}`)
```

Comment:
This is a "just in case": I think that including the `eventEnvelope` in the string like this will make it appear as `[object Object]` so, if we face this error, it will be almost impossible to know which event is too large. 
Could you please check if this is the case and, if so, do a, for example, JSON.stringify with the event envelope?